### PR TITLE
Add angular tree shaking blog

### PR DESCRIPTION
**Pull request by santoshyadavdev**
## Summary by CodeRabbit

- **New Features**
	- Added a new blog post titled "The Tree Shaking Journey in Angular: A Deep Dive" discussing tree shaking improvements in Angular.
	- Enhanced data representation for images with the addition of a `bundle` key in the media database.
	- Introduced a new optional `preview` field for blog entries to hold a preview string.
- **Improvements**
	- Updated image source selection logic and dynamic alt text for better accessibility in the blog hero component.
	- Streamlined alt text for images in blog cards and latest articles to focus solely on the post title.
	- Updated metadata for an existing blog post to reference a new image path, ensuring consistency across fields.

### CODE REVIEW

**Comment by gioboa** [comment]
Thanks for sharing it

File: src/content/blog/2024-12-20-the-tree-shaking-journey-in-angular-a-deep-dive.mdx [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+import individualmodule from '../../images/bundle/tree-shaking-modules.png';
+import standalone from '../../images/bundle/tree-shaking-standalone.png'
+import publishable from '../../images/bundle/tree-shaking-publish.png';
+import buildwarning from '../../images/bundle/build-warnign.png';
+
+
+
+Hey friends, how are you doing? This year, I tried a new Angular version and wanted to experiment with how the tree shaking works with the latest version, especially once the standalone components came into the game. The result surprised me, so let's learn what has improved and what you need to do to improve your build time and optimize your production bundle.
+
+## Whats tree shaking
```
Comment (yshashi) [inline src/content/blog/2024-12-20-the-tree-shaking-journey-in-angular-a-deep-dive.mdx:32]:
```suggestion
## What is Tree Shaking?
```

File: src/content/blog/2024-12-20-the-tree-shaking-journey-in-angular-a-deep-dive.mdx [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+Most of us use the above tools when we want our Angular application to drop dead code or do tree shaking.
+
+But what if you dont need to do this anymore? Dont need a big statement, but you only sometimes need to do secondary-entrypoint for tree shaking.
+
+> [!NOTE] 
+> Please use the standalone component by default. I am showing examples with the module if you cannot use standalone yet. This means you are not on Angular 14 or above. You will not see the tree-shaking advantages, but you can make your code future-proof, and as soon as you upgrade to Angular 16, you will get tree-shaking out of the box.
+
+
+### Lets dive into code
```
Comment (yshashi) [inline src/content/blog/2024-12-20-the-tree-shaking-journey-in-angular-a-deep-dive.mdx:65]:
```suggestion
### Let's dive into the code
```

File: src/content/blog/2024-12-20-the-tree-shaking-journey-in-angular-a-deep-dive.mdx [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+
+
+## Some closing notes
+
+1. If you are using Angular 14 or below, the exercise we did may not work to reduce your bundle size, but my recommendation is to start adopting standalone. Once you migrate to Angular 15, use `ng g @angular/core:standalone` to get the minimal code in your final production bundle.
+1. The above suggestion applies to your application and library code.
+1. For libraries with a lot of components, 
+  1. I recommend splitting them into smaller libs.
+  1. If it is publishable, I recommend going with secondary-entrypoint.
+1. With Latest Angular language service you get get warning in IDE if you are importing unused Component
```
Comment (yshashi) [inline src/content/blog/2024-12-20-the-tree-shaking-journey-in-angular-a-deep-dive.mdx:321]: We can fix the repetition of the word get

Comment (santoshyadavdev) [reply src/content/blog/2024-12-20-the-tree-shaking-journey-in-angular-a-deep-dive.mdx]: thanks

**Pull request merged**